src/clj/fluree/db/api.cljc

```diff
@@ -43,12 +42,10 @@
                (reject res)
                (resolve res))))))))
 
-;; ledger operations
-
 (defn parse-connection-options
```

Comment:
This is still a work in progress, but I think the changes I'm envisioning are big enough to warrant their own pr so I decided to leave this as is for now. 

Since the cyclic namespace dependencies have (mostly) been cleaned up, I think we should do two things to clean up the clojure api a bit more: (1) remove the "ledger" as a first class api object and (2) have the `fluree.db.connection` namespace be the api entry point (appropriately renamed, of course). 

Ledger data is already cached on the connection object, and the current ledger object is another, in my opinion redundant, source of state. We should load dbs and transact through the connection object, both localizing the application state to the connection and simplifying the api. 

We have a `fluree.db.api` namespace because the namespace hierarchy was so complex and cyclic. now that most functionality flows through the connection, the code in the connection namespace can serve as the main api with a little more cleanup, and the separate api (as well as transact and query api namespaces) would be made redundant.